roadmap/implementers-guide/src/runtime/parainherent.md

```diff
@@ -16,13 +16,19 @@ Included: Option<()>,
 
 ## Entry Points
 
-* `inclusion`: This entry-point accepts three parameters: The relay-chain parent block header, [`Bitfields`](../types/availability.md#signed-availability-bitfield) and [`BackedCandidates`](../types/backing.md#backed-candidate).
+* `enter`: This entry-point accepts three parameters: The relay-chain parent block header, [`Bitfields`](../types/availability.md#signed-availability-bitfield) and [`BackedCandidates`](../types/backing.md#backed-candidate).
     1. Hash the parent header and make sure that it corresponds to the block hash of the parent (tracked by the `frame_system` FRAME module),
+    1. Invoke `Disputes::provide_multi_dispute_data`.
+    1. If `Disputes::is_frozen`, return and set `Included` to `Some(())`.
+    1. If there are any created disputes from the current session, invoke `Inclusion::collect_disputed` with the disputed candidates. Annotate each returned core with `FreedReason::Concluded`.
```

Comment:
When disputes are initiated, we remove the block from pending availability. This allows us to roll back chains to the block before blocks are _included_ as opposed to backing. It's important to do this before processing bitfields.

Reply:
`Inclusion::collect_disputed` is kind of expensive so it's important to gate this on whether there are actually any new disputes. Which should be never.

Reply:
both make sense

Reply:
Could the ratiionales be added to the document?